backend/core/configuration.py

```diff
@@ -75,6 +76,7 @@ def __init__(self, connection: Connection) -> NoReturn:
             self.module_name = "configuration"
             self.connection = connection
             self.file = "/etc/artemis/config.yaml"
+            self.temp_file = "/etc/artemis/config.yaml.tmp"
```

Comment:
would it make more sense to use an actual temp file?
https://docs.python.org/3/library/tempfile.html

Reply:
@slowr no because the tempfile API does not offer anything in terms of atomicity. The whole reason I am doing this is because I need the atomicity that the os.rename (or move) action provides to us (because the yaml dumper is problematic in this regard in coordination with the observer). This renaming guarantees that the modification will happen "all at once".